Leuchtfeuer/mautic-housekeeping-bundle
# MTC-2953 Add the option to set the content of tokens in email_stats to NULL and the improvement that only email_stats entries will be cleaned that referenced emails entry is currently not published

## Description
This feature adds 

- the option to the leuchtfeuer housekeeping bundle to set the content of tokens fields in email_stats to NULL to "reduce the weight" of the entries and

- the improvement that only email_stats entries will be cleaned that referenced emails entry is currently not published.

Important notice: If a referenced email is ever switched back to published, the contacts will get the email again. 

## How to test
Apply the patch of this PR in a mautic testenvironment with the leuchtfeuer housekeeping bundle. 

1.) Run for example `php bin/console mautic:leuchtfeuer:housekeeping -t -d 50`

2.) After that check the table email_stats -> the content in `tokens` for all rows where date_sent is older than 50 days have been set to NULL.

3.) Run for example `php bin/console mautic:leuchtfeuer:housekeeping -m -d 50`

4.) After that check the table email_stats -> all entries are gone that referenced email_id is NULL or that referenced email is not published and that are older than 50 days.

5.) If you want to do a dry run for all that, add the option flag -r.

## Notice:
This option `--email-stats-tokens` (shortcut -t) can not be combined with any "-c", "-l" or "-m" flag in one command.
If you do so, it will throw you an error with the notice.
